File: python/19c_144A_ArrivalOfTheGeneral.py
def main():
    n = int(input())
    soldiers = input().split()
    big, small = 0, 0
    for i in range(n):
        if int(soldiers[i]) <= int(soldiers[small]) :
            small = i
        if int(soldiers[i]) > int(soldiers[big]) : 
            big = i
    ans = abs(0 - big) + abs(n - 1 - small)
    if small < big:
        ans -= 1
    print(ans)

if __name__ == "__main__":
    main()

File: cpp/19c_144A_ArrivalOfTheGeneral.cpp
#include<bits/stdc++.h>

int main(){
    int n, temp, big = 0, small = 101, big_id = 0, small_id = 0, ans;
    std::vector<int> soldiers;
    std::cin>>n;
    for (int i = 0; i < n; i++){
        std::cin>>temp;
        soldiers.push_back(temp);
        if(temp > big){
            big = temp;
            big_id = i;
        }
        if (temp <= small){
            small = temp;
            small_id = i;
        }
    }
    ans = abs(0 - big_id) + abs(n-1 - small_id);
    if (small_id < big_id){
        ans --;
    }
    std::cout<<ans;
    return 0;
}